pkg/daemon/event_recorder.go

```diff
@@ -0,0 +1,52 @@
+package daemon
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedv1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	snclientset "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/client/clientset/versioned"
+)
+
+type EventRecorder struct {
+	client           snclientset.Interface
+	node             string
+	eventRecorder    record.EventRecorder
+	eventBroadcaster record.EventBroadcaster
+}
+
+// NewEventRecorder Create a new EventRecorder
+func NewEventRecorder(c snclientset.Interface, n string, kubeclient kubernetes.Interface) *EventRecorder {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(4)
+	eventBroadcaster.StartRecordingToSink(&typedv1core.EventSinkImpl{Interface: kubeclient.CoreV1().Events("")})
```

Comment:
Oh, that's exactly what I was referring to. Thanks for explaining.

About the deletion of the events, it looks like it can be tuned via a [kubeapi-server argument](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/):

```
--event-ttl duration     Default: 1h0m0s
```

IMO it's OK that is on the cluster administrator's side